assets/javascripts/openqa.js

```diff
@@ -248,30 +253,44 @@ function restartJob(ajaxUrl, jobId) {
     }
     addFlash('danger', errorMessage);
   };
-
-  return fetchWithCSRF(ajaxUrl, {method: 'POST'})
+  const body = {jobs: jobIds, comment: comment};
+  return fetchWithCSRF(ajaxUrl, {method: 'POST', body: JSON.stringify(body)})
```

Comment:
We are posting `multipart/formdata` here, not `application/json`.
Also, the job ids are already part of the URL, so only the comment needs to be added to the formdata:
```suggestion
  const body = new FormData();
  body.append("comment", comment);
  return fetchWithCSRF(ajaxUrl, {method: 'POST', body: body})
```
As this is not catched by the tests, I suggest looking into the tests as well.

Reply:
I didnt know. To me it looked like json structure. and if I am not wrong we dont upload any file. So I wonder it this request meant to work with json as well. 
in any case I adapt your suggestion which I believe should work in either structure, right?

Reply:
Sorry, I meant `application/x-www-form-urlencoded`. That's the default content type for form submits.
And that's what's used for our API.
The *response* however is JSON.

Only for the tables API (machines, test suites, products) we are using `application/json`.